accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -0,0 +1,1212 @@
+has a representation dependency on _DV2_.
+
+It is a compile-time error if an extension type declaration has a
+representation dependency on itself.
+
+*In other words, cycles are not allowed. This ensures that it is
+always possible to find a non-extension type which is the ultimate
+representation type of any given extension type.*
+
+The *extension type erasure* of an extension type `V` is obtained by
+recursively replacing every subterm of `V` which is an extension type by
+the corresponding representation type.
+
+*Note that this extension type erasure exists, because it is a compile-time
+error to have a dependency cycle among extension type declarations.*
+
+Let
+<code>X<sub>1</sub> extends B<sub>1</sub>, .. X<sub>s</sub> extends B<sub>s</sub></code>
+be a declaration of the type parameters of a generic entity (*it could
+be a generic class, extension type, or mixin, or typedef, or function*).
+Let <code>BB<sub>j</sub></code> be the extension type erasure of
+<code>B<sub>j</sub></code>, for _j_ in _1 .. s_.
+It is a compile-time error if
+<code>X<sub>1</sub> extends BB<sub>1</sub>, .. X<sub>s</sub> extends BB<sub>s</sub></code>
+has any compile-time errors.
+
+*For example, the extension type erasure could map
+<code>X extends C<Y>, Y extends X</code> to
+<code>X extends Y, Y extends X</code>,
+which is an error.*
+
+An extension type declaration _DV_ named `Name` may declare one or
+more constructors. A constructor which is declared in an extension type
+declaration is also known as an _extension type constructor_.
+
+*The purpose of having an extension type constructor is that it bundles an
+approach for receiving or building an instance of the representation type
+of an extension type declaration _DV_ with _DV_ itself, and creating an
+expression whose static type is the extension type and whose value at run
+time is said representation object. Extension type constructor bodies can
+also be used to verify that the representation object satisfies the
+requirements for having that extension type.*
+
+The `<representationDeclaration>` works as a constructor. The optional
+`('.' <identifier>)` in the grammar is used to declare this constructor
+with a name of the form `<identifier> '.' <identifier>` *(at times
+described as a "named constructor")*. It is a constant constructor: If `e`
+is a constant expression and `V(e)` is not an error, then `V(e)` is a
+constant expression. Other constructors may be declared `const` or not,
+following the normal rules for constant constructors.
+
+A compile-time error occurs if an extension type constructor includes a
+superinitializer. *That is, a term of the form `super(...)` or
+`super.id(...)` as the last element of the initializer list.*
+
+A compile-time error occurs if an extension type constructor declares a
+super parameter. *For instance, `Name(super.x);`.*
+
+*In the body of a generative extension type constructor, the static type of
```

Comment:
non-redirecting

(but that is implicit, because a redirecting generative constructor has no body.)